### PR TITLE
Add additional test cases for module keymeta

### PR DESCRIPTION
**Pull request by lawli3t**
## Basics

These points need to be fulfilled for every PR:

- [x] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [x] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy. If not, fix **in this order**:
  - [ ] add a line in `doc/news/_preparation_next_release.md`
  - [ ] reformat the code with `scripts/dev/reformat-all`
  - [ ] make all unit tests pass
  - [ ] fix all memleaks
- [x] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

File: tests/abi/testabi_meta.c [hunk 1]
```diff
@@ -316,6 +342,22 @@ static void test_copy (void)
...
+	succeed_if (key1 = keyNew ("/", KEY_END), "could not create key");
+	succeed_if (key2 = keyNew ("/", KEY_END), "could not create key");
+
+	succeed_if (keySetMeta (key2, "mymeta", "a longer metavalue") == sizeof ("a longer metavalue"), "could not set metavalue");
+	succeed_if_same_string (keyValue (keyGetMeta (key2, "mymeta")), "a longer metavalue");
+
+	succeed_if (keyCopyMeta (key2, key1, "mymeta") == 0, "could not copy metavalue");
+
+	succeed_if (keyGetMeta (key1, "mymeta") == 0, "value of mymeta is not NULL");
+	succeed_if (keyGetMeta (key2, "mymeta") == 0, "value of mymeta has not been cleared");
```
Comment (lawli3t) [inline tests/abi/testabi_meta.c:355]: According to the documentation, the metakey should get deleted in the documentation if it is not available in source. This is **NOT** the case

Is this behaviour wrongly documented or is it a bug in the function?

Comment (markus2330) [reply tests/abi/testabi_meta.c:355]: I think you found a bug: If I explicitly copy "nothing" I would expect "nothing" to be in the resulting key.

Comment (lawli3t) [reply tests/abi/testabi_meta.c:355]: created a new issue, #3981

**Comment by markus2330** [comment]
Can you please rebase?

**Comment by lawli3t** [comment]
should be good now!

**Comment by markus2330** [comment]
@robaerd seems like the build job hangs in SSH copy. Can you maybe set some timeout? See https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-3851/9/pipeline

**Comment by markus2330** [comment]
jenkins build libelektra please

**Comment by robaerd** [comment]
> @robaerd seems like the build job hangs in SSH copy. Can you maybe set some timeout?

Doesn't seem stuck to me. The ssh upload step just takes a lot of time. However, 10 minutes just for the ssh upload is definitely too slow.
A solution for this would be to replace the upload step in which every file is uploaded individually, by creating a tarball of the doc directories, uploading it and extracting it at the destination.
If you want, I could create a PR for this.

**Comment by markus2330** [comment]
> Doesn't seem stuck to me. The ssh upload step just takes a lot of time. However, 10 minutes just for the ssh upload is definitely too slow.

The Jenkins job was still not finished after 2 days? Or did someone restart the job in between?

> A solution for this would be to replace the upload step in which every file is uploaded individually, by creating a tarball of the doc directories, uploading it and extracting it at the destination.
> If you want, I could create a PR for this.

If the ssh upload took only 10 minutes then the problem was probably somewhere else (btw. I aborted the job, how did you see how long it took?).

If the problem really was the transfer: something like rsync would be ideal.

Btw. In general it is always good if the build server jobs try to redo things if they might fail on network problems.

**Comment by robaerd** [comment]
> The Jenkins job was still not finished after 2 days? Or did someone restart the job in between?

According to the logs the [build job](https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-3851/9/pipeline) was triggered by the branch indexing trigger and started on 20.09 at 12:50. Maybe you mistook the currently running job for the previous one that was triggered on 16.09 by a commit?

> ... (btw. I aborted the job, how did you see how long it took?).

You can check the duration of each step + stage in the [Blue Ocean UI](https://build.libelektra.org/blue/organizations/jenkins/libelektra/detail/PR-3851/9/pipeline/473) or get the overall timing info from the [classic UI](https://build.libelektra.org/job/libelektra/job/PR-3851/9/).

**Comment by lawli3t** [comment]
jenkins build libelektra please

**Comment by lawli3t** [comment]
LGTM now! :rocket:

**Comment by mpranj** [comment]
Thank you so much!

**Pull request merged**